File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
N = int(input())
bag = N // 5
left = N % 5

while True:
    if left > N:
        bag = -1
        break
    elif left % 3 == 0:
        bag += left // 3
        break
    else:
        left += 5
        bag -= 1

print(bag)

File: 백준/Silver/2839. 설탕 배달/설탕 배달.c
#include <stdio.h>

int main(){
    int N;
    scanf("%d",&N);
    int bag = N/5;
    int left = N%5;

    while(1){
        if (left > N){
            bag = -1;
            break;
        }
        else if (left % 3 == 0){
            bag += left / 3;
            break;
        }
        else{
            left += 5;
            bag--;
        }
    }

    printf("%d\n",bag);
}